### PR TITLE
Migrate to the new config

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/18119

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -80,7 +79,6 @@ func Parse() (Config, error) {
 
 	kingpin.Flag("address", "The address to listen on").Default(defaultAddress).StringVar(&config.Address)
 	kingpin.Flag("admin-group", "Tenant Admin Target Group").Required().StringVar(&config.AdminGroup)
-	kingpin.Flag("all-target-group", "View All Target Group").Required().StringVar(&config.AllTargetGroup)
```
Comment (kopiczko) [inline config/config.go:83]: Apparently not a thing anymore.

https://gigantic.slack.com/archives/C7865GLSY/p1626876882006500

**Comment by kopiczko** [comment]
I won't deal with nancy in this PR.

File: helm/aws-admission-controller/templates/deployment.yaml [hunk 2]
```diff
@@ -32,34 +32,32 @@ spec:
...
+              value: {{ .Values.workloadCluster.kubernetes.api.clusterIPRange }}
+            - name: DEFAULT_KUBERNETES_ENDPOINT
+              value: {{ .Values.workloadCluster.baseDomain }}
             - name: DEFAULT_AWS_INSTANCE_TYPES
-              value: {{ join "," .Values.Installation.V1.Provider.AWS.EC2.Instance.Allowed }}
+              value: {{ join "," .Values.aws.instance.allowed }}
             - name: DEFAULT_AWS_REGION
-              value: {{ .Values.Installation.V1.Provider.AWS.Region }}
+              value: {{ .Values.aws.region }}
+            {{- $workloadClusterCNICIDRSplit := splitn "/" 2 .Values.workloadCluster.cni.cidr }}
```
Comment (calvix) [inline helm/aws-admission-controller/templates/deployment.yaml:53]: why is the split here?

Comment (calvix) [reply helm/aws-admission-controller/templates/deployment.yaml:53]: is it just to save variables and have 2 values in one?

**Review by njuettner — APPROVED** [review]
lgtm

File: helm/aws-admission-controller/templates/deployment.yaml (hunk 2 above)
Comment (kopiczko) [reply helm/aws-admission-controller/templates/deployment.yaml:53]: Yeah, this `.Values.Installation.V1.Guest.Calico.CIDR` and `.Values.Installation.V1.Guest.Calico.Subnet` makes zero sense to me. Now you have `.Values.workloadCluster.cni.cidr` in values.yaml. It's just internally weird, but I don't want to deal with that. You can address that any time if you want to.

**Review by calvix — APPROVED** [review]

**Pull request merged**